File: BOrganizer/BOrganizer.InertiaReact/ClientApp/src/js/types/payment.ts
export type Payment = {
    paymentId: number;
    date: string;
    sender: string;
    receiver: string;
    product: string;
    currency: string;
    method: string;
    brutto: number;
    notes?: string;
    rechnungId?: number;
};

export interface PaymentDto {
    paymentId?: number;
    date: string;
    netto: number;
    mwstPercent: number;
    mwstEuro: number;
    brutto: number;
    sender: string;
    receiver: string;
    product: string;
    currency: string;
    method: string;
    notes?: string;
    rechnungId?: number;
    senderId?: number;
    receiverId?: number;
}

File: BOrganizer/Rechnungen/Model/Payments/Payment.cs
namespace Rechnungen.Model.Payments;

public record Payment(
    DateOnly Date,
    decimal MwstPercent,
    decimal Netto,
    decimal Brutto,
    decimal MwstTotal,
    string Sender,
    string Receiver,
    string Product,
    string Currency,
    string? Method = null,
    string? Notes = null,
    long? SenderId = null,
    long? ReceiverId = null,
    long? RechnungId = null
)
{
    public required long? Id { get; init; }
}